CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+cmake_minimum_required(VERSION 3.15.7)
+
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+include(toolchain.cmake) # TODO(melvinw): this should not be included here. It should be configured by ide or by passing it to cmake
+
+# Toolchain must be included before project declaration
+project(MarsRoverFirmware)
+
+# Global Settings
+set(COMPILE_FLAGS
+        -MMD
+        -MP
+
+        -O3
+        -g
+
+        -Wall
+        -Wextra
+        -Wpedantic
+        -Wvla
+        -Werror
+
+        -Wno-unused-parameter
+        -Wno-missing-field-initializers
+
+        -ffunction-sections
+        -fdata-sections
+        -fomit-frame-pointer
+        -funsigned-char
+
+        -fno-exceptions
+        -fno-delete-null-pointer-checks
+
+        -fstack-protector-strong
+
+        $<$<COMPILE_LANGUAGE:ASM>:-xassembler-with-cpp>
+        $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
+        )
+add_compile_options(${PROCESSOR_FLAGS} ${COMPILE_FLAGS})
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_CXX_STANDARD 20)
+
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Include mbed-os
+include(mbed-os.cmake)
+
+# Include library CMakeLists
+add_subdirectory(lib/can)
+add_subdirectory(lib/controller-inputs)
+add_subdirectory(lib/controllers)
+add_subdirectory(lib/neopixel)
+add_subdirectory(lib/pid)
+add_subdirectory(lib/sensors)
+add_subdirectory(lib/servo)
+
+# TODO(wmmc88): check if its in supported configuration list and warn if not in list. This should be in wrapper makefile
+if (NOT DEFINED APP)
+    message(FATAL_ERROR "APP variable not set in CACHE. Please invoke CMake with \"-DAPP=<app-name>\"")
```

Comment:
with the previous makefile, if a wrong app or target was selected, we'd give an error message and list out the available apps. could we do this here too.

Reply:
I'll add the list and move this to a makefile wrapper